addons/project/models/project_task.py

```diff
@@ -944,6 +949,26 @@ def create(self, vals_list):
                 task.message_subscribe(follower.partner_id.ids, follower.subtype_ids.ids)
             if current_partner not in task.message_partner_ids:
                 task.message_subscribe(current_partner.ids)
+            if task.email_cc:
+                partners = self.env['res.partner'].search([('email', 'in', tools.email_split(task.email_cc))])
+                partners_with_internal_user_ids = set()
+                for partner in partners:
+                    for user in partner.user_ids:
+                        if user.has_group('base.group_user'):
+                            partners_with_internal_user_ids.add(partner.id)
+                            break
```

Comment:
We should use `share` field to know if the user is an internal user or not. 🙂 
```suggestion
                for partner in partners:
                    if not all(u.share for u in partner.user_ids):
                        partners_with_internal_user_ids.add(partner.id)
```